chore: Update baselines before tag release 1.9.0 (#1645)
patternfly/patternfly-elements: elements/pfe-readtime/docs/index.md

---
layout: layout-basic.html
title: Readtime
description: Takes in the word count of a section and calculates the estimated read time based on language
package: pfe-readtime
packages:

>   - pfe-readtime

Author: ```suggestion
  - pfe-readtime
  - pfe-card
```

tags:
  - component
---

::: section header
# {{ title }}
:::

::: section
## Overview

This component takes in the word count of a given section and does a calculation on that number to return an estimated read time based on language. The words-per-minute values were sourced from [this article](https://irisreading.com/average-reading-speed-in-various-languages) with the data originating from [this research](https://iovs.arvojournals.org/article.aspx?articleid=2166061). For more information, see the [Readtime calculation information](#readtime-calculation).

> <pfe-readtime word-count="500" hidden>%t-minute readtime</pfe-readtime>

Author: ```suggestion
<pfe-card border><pfe-readtime word-count="500" hidden>%t-minute readtime</pfe-readtime></pfe-card>
```

:::

::: section
## Installation

```shell
npm install @patternfly/{{ package }}
```
:::

::: section
## Usage

```html
    <pfe-readtime for="#readtime1" hidden>%t-minute readtime</pfe-readtime>
    <pfe-readtime word-count="1200" hidden>%t-minute readtime</pfe-readtime>
```
:::

::: section
## Accessibility
This component functions purely as inline-content and does not require any focus state.  Should be read by screen-readers inline with it's contextual content.
:::

::: section
## Attributes

> - `word-count`:  Is the data-attribute you will leverage for readtime. Example: word-count="2500" will let the component know that there is 2500 words in that section and it will do it's calculations based on that number. If you don't want to/have a data-attribute to leverage you can use id="readtime1" and it will get the word count for you.

Author: ```suggestion
- `word-count`: allows you to manually set the number of words to use in the readtime calculations. Example: `word-count="2500"` will set the component to use 2500 words in it's calculations.
```

> -`wpm`: Is the attribute used to store the average words per minute readtime for each supported country. For more information on these you can read [https://irisreading.com/average-reading-speed-in-various-languages](https://irisreading.com/average-reading-speed-in-various-languages and https://iovs.arvojournals.org/article.aspx?articleid=216606) and [https://iovs.arvojournals.org/article.aspx?articleid=216606](https://iovs.arvojournals.org/article.aspx?articleid=216606).

Author: ```suggestion
-`wpm`: Is the attribute used to store the average words per minute readtime for each supported country. For more information on these you can read [https://irisreading.com/average-reading-speed-in-various-languages](https://irisreading.com/average-reading-speed-in-various-languages) and [https://iovs.arvojournals.org/article.aspx?articleid=216606](https://iovs.arvojournals.org/article.aspx?articleid=216606).
```
